From c65b2674cbdbf3552d697babbdd5e6ab7c2e5415 Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@windriver.com>
Date: Sat, 29 Jun 2019 14:25:21 +0800
Subject: [PATCH] support to get kickstart from network

We support http/https/ftp protocol to get kickstart file

Upstream-Status: Inappropriate [oe specific]

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>

Update for anaconda 32.

Signed-off-by: Kai Kang <kai.kang@windriver.com>

---
 pyanaconda/startup_utils.py | 13 ++++++++++---
 1 file changed, 10 insertions(+), 3 deletions(-)

diff --git a/pyanaconda/startup_utils.py b/pyanaconda/startup_utils.py
index bff65308e..3a4ba77d9 100644
--- a/pyanaconda/startup_utils.py
+++ b/pyanaconda/startup_utils.py
@@ -355,8 +355,11 @@ def find_kickstart(options):
     :param options: command line/boot options
     :returns: a path to a kickstart file or None
     """
-    if options.ksfile and not options.liveinst:
-        if not os.path.exists(options.ksfile):
+    if options.ksfile:
+        if not os.path.exists(options.ksfile) and \
+           not options.ksfile.startswith("http://") and \
+           not options.ksfile.startswith("https://") and \
+           not options.ksfile.startswith("ftp://"):
             stdout_log.error("Kickstart file %s is missing.", options.ksfile)
             util.ipmi_report(constants.IPMI_ABORTED)
             sys.exit(1)
@@ -373,8 +376,12 @@ def find_kickstart(options):
     else:
         ks_files = ["/usr/share/anaconda/interactive-defaults.ks"]
 
+    log.info("ks_files: %s" % ks_files)
     for ks in ks_files:
-        if not os.path.exists(ks):
+        if not os.path.exists(ks) and \
+           not ks.startswith("http://") and \
+           not ks.startswith("https://") and \
+           not ks.startswith("ftp://"):
             continue
 
         return ks
